src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:      i.eodag
+#
+# AUTHOR(S):   Hamed Elgizery
+# MENTOR(S):   Luca Delucchi, Veronica Andreo, Stefan Blumentrath
+#
+# PURPOSE:     Downloads imagery datasets e.g. Landsat, Sentinel, and MODIS
+#              using EODAG API.
+# COPYRIGHT:   (C) 2024-2025 by Hamed Elgizery, and the GRASS development team
+#
+#              This program is free software under the GNU General Public
+#              License (>=v2). Read the file COPYING that comes with GRASS
+#              for details.
+#
+#############################################################################
+
+
+# %Module
+# % description: Eodag interface to install imagery datasets from various providers.
+# % keyword: eodag
+# % keyword: sentinel
+# % keyword: landsat
+# % keyword: modis
+# % keyword: datasets
+# % keyword: download
+# %end
+
+# %option
+# % key: dataset
+# % type: string
+# % description: Imagery dataset to search for
+# % required: yes
+# % answer: S1_SAR_GRD
+# % guisection: Filter
+# %end
+
+# %option G_OPT_M_DIR
+# % key: output
+# % description: Name for output directory where to store downloaded data OR search results
+# % required: no
+# % guisection: Output
+# %end
+
+# %option
+# % key: id
+# % type: string
+# % multiple: yes
+# % description: List of scenes IDs to download
+# % guisection: Filter
+# %end
+
+# %option
+# % key: provider
+# % type: string
+# % description: Provider to use for searching/downloading, if none is provided the searching will be done according to the config file priority
+# % required: no
+# % guisection: Filter
+# %end
+
+# %option
+# % key: start
+# % type: string
+# % description: Start date ('YYYY-MM-DD'), by default it is 60 days ago
+# % guisection: Filter
+# %end
+
+# %option
+# % key: end
+# % type: string
+# % description: End date ('YYYY-MM-DD')
+# % guisection: Filter
+# %end
+
+# %option
+# % key: relation
+# % type: string
+# % description: Relation with area of interest
+# % options: intersects, contains, within
+# % answer: intersects
+# % guisection: Optional
+# %end
+
+# %option
+# % key: clouds
+# % type: integer
+# % description: Maximum cloud cover percentage for Landsat scene
+# % required: no
+# % guisection: Filter
+# %end
+
+# %option
+# % key: timeout
+# % type: integer
+# % description: Download timeout in seconds
+# % answer: 300
+# % guisection: Optional
+# %end
+
+
+# %flag
+# % key: l
```

Comment:
We may consider a format option (with standard parser option G_OPT_M_FORMAT) for all printing (CSV, JSON, plain, ...)

Reply:
Handled further here: https://github.com/HamedElgizery/grass-addons/issues/4